### PR TITLE
feat: a.di add events to notice, add tests

### PR DESCRIPTION
**Pull request by BATMAH69**
add events
add tests

### CODE REVIEW

File: bsc-adi-governance/src/services/cross-chain-controller/CrossChainController.srv.ts [hunk 1]
```diff
@@ -44,6 +48,70 @@ export class CrossChainControllerSrv {
     return this.name
   }
 
+  public async handleBlock(block: BlockEvent): Promise<Finding[]> {
+    const start = new Date().getTime()
+    const findings: Finding[] = []
+
+    const [envelopeDelayFindings] = await Promise.all([this.handleEnvelopeDelay(block)])
+    findings.push(...envelopeDelayFindings)
```
Comment (Ivan-Feofanov) [inline bsc-adi-governance/src/services/cross-chain-controller/CrossChainController.srv.ts:56]:
```suggestion
    const findings = await this.handleEnvelopeDelay(block)
```

File: bsc-adi-governance/src/services/cross-chain-controller/CrossChainController.srv.ts [hunk 2]
```diff
@@ -44,6 +48,70 @@ export class CrossChainControllerSrv {
...
+
+    this.logger.info(elapsedTime(this.name + '.' + this.handleBlock.name, start))
+    return findings
+  }
+
+  private async handleEnvelopeDelay(block: BlockEvent): Promise<Finding[]> {
+    const start = new Date().getTime()
+    const out: Finding[] = []
+
+    if (block.blockNumber % PERIODICAL_BLOCK_INTERVAL != 0) {
```
Comment (Ivan-Feofanov) [inline bsc-adi-governance/src/services/cross-chain-controller/CrossChainController.srv.ts:66]: Maybe it'd be reasonable to move this check to the beginning of the block?

Comment (BATMAH69) [reply bsc-adi-governance/src/services/cross-chain-controller/CrossChainController.srv.ts:66]: In current bots code there are examples of different sequences of this 3 lines ('start', 'out' & '%') so any sequence takes place in existence.
It seems to me a good solution to leave `start` at the beginning of the function to make it easier to control. And `out` is declared for a uniform return from the function. 
If this is critical, I suggest adding this to the bot template so as not to solve it separately in each specific case.

**Review by Ivan-Feofanov — APPROVED** [review]
LGTM

**Pull request merged**